queries/nix.scm

```diff
@@ -0,0 +1,146 @@
+;; Statements
+
+[
+    (binding)
+] @statement
+;; This matches assert statements as stand-alone statements. This
+;; is because it's common to have a series of assert statements in a file, which
+;; would otherwise all match as a single statement. See the playground/nix/asserts.nix
+;; file for an example.
+(assert_expression
+    "assert" @statement.start
+    condition: (_)
+    ";" @statement.end
+    body: (_)
+)
+(binding_set) @map.iteration @list.iteration @ifStatement.iteration
+(source_code) @statement.iteration @map.iteration @comment.iteration
+
+;;!! let
+;;!!   a = 1;
+;;!    <*****
+;;!!   b = 2;
+;;!    *****>
+;;!! in a + b
+(let_expression
+    "let"
+    (_) @statement.iteration @value.iteration @name.iteration
+)
+
+;; Conditionals
+
+;;!! key = if a then b else c;
+;;!        ^^^^^^^^^^^^^^^^^^
+(if_expression) @ifStatement @branch.iteration
+;;!! key = if a then b else c;
+;;!        ^^^^
+(if_expression
+    "if" @branch.start
+    condition: (_) @branch.end
+)
+;;!! key = if a then b else c;
+;;!             ^^^^^^
+(if_expression
+    "then" @branch.start
+    consequence: (_) @branch.end
+)
+;;!! key = if a then b else c;
+;;!                    ^^^^^^
+(if_expression
+    "else" @branch.start
+    alternative: (_) @branch.end
+)
+
+;; Lists and maps
+
+;;!! foo = [ a b c ];
+;;!        ^^^^^^^^^
+(list_expression) @list
+
+;;!! foo = { x = 1; y = 2; };
+;;!        ^^^^^^^^^^^^^^^^^^
+[
+    (attrset_expression)
+    (rec_attrset_expression)
+] @map @statement.iteration @value.iteration @name.iteration
+
+;;!! foo = { x = 1; y = 2; };
+;;!          ^
+(attrpath) @collectionKey
+
+;; Strings
+
+;;!! # foo
+;;!  ^^^^^
+(comment) @comment @textFragment
```

Comment:
if you're defining text fragments, it's worth having a `"change round"` test inside a string and comment

Reply:
Done for comment, so will mark as resolved. Strings have to work first, before I add that test though :D

Reply:
Have you tried "change round" inside a string? It's independent of the "string" scope used when you say "take string"

Reply:
Reopening this one to see whether "change round" works inside a string. The machinery to make that work is independent of the "string" scope used when you say "take string", so this discussion is orthogonal to https://github.com/cursorless-dev/cursorless/pull/1911#discussion_r1343335701, which is presumably the thread you marked this one as resolved in favour of

Reply:
Just to clarify, do you mean try change round in a string with the cursor outside of the () or inside? I did test it today, and it works inside a string only if the cursor is inside (), and doesn't work if outside the (). I believe this is consistent with how it works with comment, but I'll have to confirm tomorrow.

Reply:
Just tested again and for comments and strings of both types (`"` and `''`) you can `change round` with cursor inside parens and it will work (delete the `(...`) and place cursor there). But you can't do it with the cursor outside of the parens. I tested other languages and this seems to be consistent. For some reason I thought cursorless would jump to the closes `()` pair on the line if the cursor was outside, but I guess I'm thinking vim...

Reply:
Yeah you're thinking of vim 😄. What you've described is functioning as I'd expect